File: patterns/manuelle-trigger-erkennung/rules-python/manuelle_trigger_erkennung_timestamp.py
from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger


@rule(triggers=[ItemStateChangeTrigger("vTimeOfDay")])
class SetLightsBasedOnTimeOfDay:
    def execute(self, module, input):
        Registry.getItem("gLights_WEATHER_OVERRIDE").postUpdate("OFF")

        time_of_day = str(Registry.getItem("vTimeOfDay").getState())

        off_group = next(
            (g for g in Registry.getItem("gLights_OFF").getMembers()
             if g.getName() == "gLights_OFF_" + time_of_day),
            None,
        )
        if off_group:
            for light in off_group.getMembers():
                if str(light.getState()) != "OFF":
                    light.sendCommand("OFF")

        on_group = next(
            (g for g in Registry.getItem("gLights_ON").getMembers()
             if g.getName() == "gLights_ON_" + time_of_day),
            None,
        )
        if on_group:
            for light in on_group.getMembers():
                if str(light.getState()) != "ON":
                    light.sendCommand("ON")

        self.logger.info("vTimeOfDay=" + time_of_day + " -> Lichter angepasst")

File: patterns/manuelle-trigger-erkennung/rules-js/manuelle-trigger-erkennung-timestamp.js
rules.JSRule({
  name: "Set lights based on Time of Day",
  triggers: [triggers.ItemStateChangeTrigger("vTimeOfDay")],
  execute: (event) => {
    items.getItem("gLights_WEATHER_OVERRIDE").postUpdate("OFF");

    const timeOfDay = items.getItem("vTimeOfDay").state;

    const offGroup = items.getItem("gLights_OFF").members.find((g) => g.name === "gLights_OFF_" + timeOfDay);
    if (offGroup) {
      offGroup.members.filter((l) => l.state !== "OFF").forEach((l) => {
        l.sendCommand("OFF");
        console.log(l.name + " -> OFF");
      });
    } else {
      console.warn("Keine OFF-Gruppe fuer " + timeOfDay + " gefunden");
    }

    const onGroup = items.getItem("gLights_ON").members.find((g) => g.name === "gLights_ON_" + timeOfDay);
    if (onGroup) {
      onGroup.members.filter((l) => l.state !== "ON").forEach((l) => {
        l.sendCommand("ON");
        console.log(l.name + " -> ON");
      });
    } else {
      console.warn("Keine ON-Gruppe fuer " + timeOfDay + " gefunden");
    }
  }
});
